### ISSUE TITLE
Memory Leak when the component is destroyed but the MutationObserver's callback is never called

### ISSUE THREAD

**Issue by LazaroOnline** [pqina/vue-filepond#100]
### Summary
Memory Leak when the component is destroyed but the detached callback is never called.
It happens in some cases when "vue-filepond.js" "detached()" method is never called even if the "destroyed()" method for the wrapper was called.
The MutationObserver's callback is never called, I guess that observer doesn't cover all cases.

### How to reproduce
Using Vuetify's data table with expandable rows that contains a FilePond instance, on closing the expandable row the contents are destroyed by Vue, but FilePond fails to destroy it's _pond instance.
https://codesandbox.io/s/vue-filepond-memory-leak-gh6yf
See also:
https://v15.vuetifyjs.com/en/components/data-tables

### Expected behaviour
the method "this._pond.destroy()" should be called on the component when it is destroyed, and if there is any reason to wait for it, it should make sure it always end up firing, otherwise the memory is never released!
If no better way to receive a callback is found, then I would suggest just calling this._pond.destroy() on the component's destroy method, or create a property to allow destroying it directly.

| Environment  | Version
|------------- | -----------
| OS           |  Windows 10
| Device       | Microsoft laptop
| Browser      | Chrome 75.0.3770.142 (Official Build) (64-bit)

**Comment by rikschennink** [comment]
Thanks for reporting, will take a look as soon as possible. The MutationObserver is there to allow removal after a fade out transition.

```js
destroyed() {
    console.log("Uploader component destroyed... but FilePond was not destroyed with _pond.destroy()");
    // The FilePond VueComponent is also destroyed...
    // But the vainilla "FilePond" instance is not destroyed!
    // it is waiting for some event to fire that will never fire in this example.
    // I guess FilePond didn't account for this scenario.
    // but this is valid! Vuetify properly destroys the objects after closing the expandable.
  }
```

You're logging that it is not destroyed but how have you determined this?

**Comment by LazaroOnline** [comment]
I determined that the method was not being called by adding a console.log inside the "detached" method of "vue-filepond.js".
I updated the code sandbox example to add those console logs so it is easier to check.
It can also be detected by profiling the memory usage in chrome and observing if the memory is being released after opening and closing the row by clicking it.

**Comment by rikschennink** [comment]
@LazaroOnline Should be fixed in `5.1.3`, the MutationObserver now watches the document root for changes

**Comment by LazaroOnline** [comment]
Yay! It does work now in all my tests is calling the destroy method properly.
Great job and fast response, Thank you very much!
